### PR TITLE
Badge fixes

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Link to journey badges page from dashboard
- Fix invalid name
- Order badges by rarity

### CODE REVIEW

File: app/controllers/dashboard_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class DashboardController < ApplicationController
   def show
     @user_tracks = current_user.user_tracks.order(last_touched_at: :desc).limit(3)
     @num_user_tracks = current_user.user_tracks.count
-    @featured_badges = Badge.where(id: current_user.acquired_badges.revealed.order('id desc').limit(4).pluck(:badge_id))
+    @featured_badges = Badge.where(id: current_user.acquired_badges.revealed.pluck(:badge_id)).order(Arel.sql("FIND_IN_SET(rarity, 'legendary,ultimate,rare,common')")).limit(4) # rubocop:disable Layout/LineLength
```
Comment (iHiD) [inline app/controllers/dashboard_controller.rb]:
```suggestion
    @featured_badges = Badge.where(id: current_user.acquired_badges.revealed.select(:badge_id)).order(Arel.sql("FIND_IN_SET(rarity, 'legendary,ultimate,rare,common')")).limit(4) # rubocop:disable Layout/LineLength
```

I think you want select on the first to use an inner query. I'd break it over 2 lines too.

File: app/models/badge.rb [hunk 2]
```diff
@@ -1,10 +1,12 @@
 class Badge < ApplicationRecord
   has_many :acquired_badges, class_name: "User::AcquiredBadge", dependent: :destroy
 
-  RARIRIES = %i[common rare ultimate legendary].freeze
+  RARITIES = %i[common rare ultimate legendary].freeze
```
Comment (ErikSchierboom) [inline app/models/badge.rb:4]: Fixed a small typo

File: app/models/user.rb [hunk 3]
```diff
@@ -192,9 +192,8 @@ def has_badge?(slug)
     acquired_badges.where(badge_id: badge.id).exists?
   end
 
-  # TODO: Order by rarity
```
Comment (ErikSchierboom) [inline app/models/user.rb:195]: Jay. Removed a todo

File: test/models/user_test.rb [hunk 4]
```diff
@@ -116,6 +116,42 @@ class UserTest < ActiveSupport::TestCase
...
+    ultimate_badge_1 = create :lackadaisical_badge
+    legendary_badge_1 = create :begetter_badge
+    legendary_badge_2 = create :moss_badge
+
+    create :user_acquired_badge, user: user, badge: common_badge_1
+    create :user_acquired_badge, user: user, badge: common_badge_2
+    create :user_acquired_badge, user: user, badge: rare_badge_1
+    create :user_acquired_badge, user: user, badge: ultimate_badge_1
+    create :user_acquired_badge, user: user, badge: legendary_badge_1
+    create :user_acquired_badge, user: user, badge: legendary_badge_2
```
Comment (iHiD) [inline test/models/user_test.rb:133]: Can you reorder these so that the ordering is more strongly tested pls.

File: test/models/user_test.rb [hunk 5]
```diff
@@ -116,6 +116,42 @@ class UserTest < ActiveSupport::TestCase
...
+
+    common_badge = create :rookie_badge
+    rare_badge = create :supporter_badge
+    ultimate_badge = create :lackadaisical_badge
+    legendary_badge = create :begetter_badge
+
+    create :user_acquired_badge, revealed: false, user: user, badge: common_badge
+    create :user_acquired_badge, revealed: false, user: user, badge: rare_badge
+    create :user_acquired_badge, revealed: false, user: user, badge: ultimate_badge
+    create :user_acquired_badge, revealed: false, user: user, badge: legendary_badge
```
Comment (iHiD) [inline test/models/user_test.rb:148]: Same with this.

**Review by iHiD — APPROVED** [review]

**Pull request merged**